### ISSUE TITLE
Mdlint: TOP004 error with wildcardRegex scope

### ISSUE THREAD

**Issue by MaoShizhong** [TheOdinProject/curriculum#27496]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

`TOP004` involves testing `wildcardRegex` against headings.
Currently, the variable is declared within a callback, on line 102. The variable is then used on lines 150 and 152 but only after leaving the callback's scope.

This results in a linting error `wildcardRegex is not defined` - tested on several project files, all of which either went straight from `### Introduction` to `### Assignment` and/or had no heading/section following `### Assignment`.
i.e. the error only triggers if there isn't at least one wildcard heading *in both* wildcard positions. e.g. `javascript/computer_science/project_linked_lists.md` passes as both wildcard positions have at least one heading (additional headings also pass).

Moving the regex declaration on line 102 to line 96 to solve the scoping issue results in a different error instead:
`javascript/computer_science/project_knights_travails.md:47 TOP004/lesson-headings Required heading structure [Context: "*"]`
This affects the same project files as the former scoping issue i.e. project files that passed the previous also pass with this change.

### Path

Other / NA

### Lesson Url

https://github.com/TheOdinProject/curriculum/blob/main/markdownlint/TOP004_lessonHeadings/TOP004_lessonHeadings.js
